docs(change): Don't mention non-existent workspace.badges (#14042)
rust-lang/cargo: CHANGELOG.md

### Changed

- ❗️ manifest: Disallow `[badges]` to inherit from `[workspace.package.badges]`.

>   This was considered a bug.
>   Keep in mind that `[badges]` is effectively deprecated.

Reviewer: What about this?

```suggestion
  The support of it was considered a bug, since `[badges]` is deprecated.
```

Either is fine. `r=me` when ready.

Author: That implies a causation which isn't there.
